src/main/java/com/thebookcooper/dao/BookInfoDAO.java

```diff
@@ -0,0 +1,76 @@
+package com.thebookcooper.dao;
+
+import com.thebookcooper.model.*;
+import com.thebookcooper.util.DataAccessObject;
+
+import java.sql.*;
+
+
+public class BookInfoDAO extends DataAccessObject<Book> {
+
+    private static final String GET_ONE = "SELECT book_id, title, isbn, publish_date, author, genre " +
+            "FROM book_info WHERE book_id=?";
+
+    private static final String INSERT = "INSERT INTO book_info (title, isbn, publish_date, author, genre) " +
+            "VALUES (?, ?, ?, ?, ?)";
+
+    public BookInfoDAO(Connection connection) {
+        super(connection);
+    }
+
+
+    @Override
+    public Book findById(long id) {
+        Book book = new Book();
+        try (PreparedStatement statement = this.connection.prepareStatement(GET_ONE)) {
+            statement.setLong(1, id);
+            ResultSet rs = statement.executeQuery();
+            while (rs.next()) {
```

Comment:
shouldnt we use a if statment here instead of a while loop.  
@Override
public Book findById(long id) {
    Book book = null; // Initialize to null to handle case where no book is found
    try (PreparedStatement statement = this.connection.prepareStatement(GET_ONE)) {
        statement.setLong(1, id);
        ResultSet rs = statement.executeQuery();
        if (rs.next()) { // Use if instead of while since only one record is expected
            book = new Book(); // Instantiate the book only if a record is found
            book.setBookId(rs.getLong("book_id"));
            book.setTitle(rs.getString("title"));
            // Corrected the data type for ISBN to String, assuming ISBNs are stored as VARCHAR in your schema
            book.setISBN(rs.getString("isbn")); 
            book.setPublishDate(rs.getDate("publish_date"));
            book.setAuthor(rs.getString("author"));
            book.setGenre(rs.getString("genre"));
        }
    } catch (SQLException e) {
        e.printStackTrace();
        throw new RuntimeException(e);
    }
    return book; // Returns null if no book is found, otherwise returns the found book
}

Reply:
yeah you're right it is only one loop no matter what since there's only one unique id. But I think its fine to leave it since it doesn't change time complexity.